bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/Workbench.java

```diff
@@ -687,8 +688,8 @@ private static void setRescaleAtRuntimePropertyFromPreference() {
 					+ " is configured (e.g., via the INI), but the according preference should be preferred instead." //$NON-NLS-1$
 			));
 		} else {
-			boolean rescaleAtRuntime = PrefUtil.getAPIPreferenceStore()
-					.getBoolean(IWorkbenchPreferenceConstants.RESCALING_AT_RUNTIME);
+			boolean rescaleAtRuntime = ConfigurationScope.INSTANCE.getNode(WorkbenchPlugin.PI_WORKBENCH)
+					.getBoolean(IWorkbenchPreferenceConstants.RESCALING_AT_RUNTIME, false);
```

Comment:
should this possibly take a system property into account for the default value?

Reply:
Question would be which one to take into account. What the preference currently does is to set a system property which configures the HiDPI behavior of SWT. If you specify that system property manually (via INI/VM argument), this will be used anyway (in the if-block above) and the preference will be ignored.
So if we want to take a preference into account, we should probably just remove the if/else and integrate the system property value into this preference evaluation. The reason for not doing it that way was that we wanted to log a warning, as the interaction between system property and preference may be hard to understand, so setting a VM argument or INI flag to define it should be avoided in favor of simply using the preference.

Reply:
So I would keep as is for now (it conforms to the existing behavior also defaulting to "false"). Additionally, this is currently only an experimental preference (which is reasonably set to "false" by default) to be replaced by a "proper" one once the feature is ready to be used productively and potentially activated by default.